File: src/dotnet/Common/Models/Orchestration/GatewayCompletionRequest.cs
using FoundationaLLM.Common.Models.Chat;
using System.Text.Json.Serialization;

namespace FoundationaLLM.Common.Models.Orchestration;

/// <summary>
/// The completion request object.
/// </summary>
public class GatewayCompletionRequest : CompletionRequest
{
    [JsonPropertyName("operation_id")]
    public string? OperationId { get; set; }

    /// <summary>
    /// The token count
    /// </summary>
    [JsonPropertyName("tokens_count")]
    public int TokensCount { get; init; }
}

File: src/python/PythonSDK/foundationallm/models/orchestration/gateway_completion_request.py
import uuid
from typing import List, Optional
from pydantic import BaseModel
from foundationallm.models.orchestration import MessageHistoryItem, OrchestrationSettings, CompletionRequestBase

class GatewayCompletionRequest(CompletionRequestBase):
    """
    Gateway completion request.
    """
    tokens_count : Optional[int] = None
